jolt-core/src/subprotocols/optimization.rs

```diff
@@ -0,0 +1,761 @@
+                    }
+
+                    let r_cycle_val = r_cycle[j_idx];
+
+                    C_summands[0] = r_cycle_val;
+                    C_summands[1] = F::one() - r_cycle_val;
+                }
+
+                // Compute eq(r_round, w_1, ..., w_{idx - 1}, c, b) for each c and b = 0, 1
+                // let eq_evals_at_idx = eval_points
+                //     .iter()
+                //     .map(|c| (((F::one() - c) * C_summands[1], *c * C_summands[0])))
+                //     .collect::<Vec<(F, F)>>();
+
+                // Evaluate eq(r_round, w_1, ..., w_{idx - 1}, c) at c = 0, 2
+                let eq_evals_at_idx = [
+                    (C_summands[1], F::zero()),
+                    (-C_summands[1], C_summands[0] + C_summands[0]),
+                ];
+
+                let univariate_poly_evals = before_idx_evals
+                    .par_iter_mut()
+                    .take(size)
+                    .zip(after_idx_evals.par_iter_mut().take(size))
+                    .enumerate()
+                    .map(|(j, (before_idx_eval, after_idx_evals))| {
+                        // let at_idx_evals =
+                        // mle_vec[D - d - 1].sumcheck_evals(j, 2, BindingOrder::HighToLow);
+                        let at_idx_evals = [
+                            mle_vec[D - d - 1].get_bound_coeff(j),
+                            mle_vec[D - d - 1].get_bound_coeff(j + mle_vec[D - d - 1].len() / 2)
+                                + mle_vec[D - d - 1]
+                                    .get_bound_coeff(j + mle_vec[D - d - 1].len() / 2)
+                                - mle_vec[D - d - 1].get_bound_coeff(j),
+                        ];
+
+                        let eq_eval_after_idx = if j_idx < T.log_2() - 1 {
+                            E_table[j_idx].get_coeff(j)
+                        } else {
+                            F::one()
+                        };
+
+                        if d > 0 {
+                            *before_idx_eval = before_idx_eval
+                                .mul_1_optimized(mle_vec[D - d - 1].get_bound_coeff(j));
+                        }
+
+                        let temp = *before_idx_eval * eq_eval_after_idx * C;
+
+                        [
+                            {
+                                let factor = at_idx_evals[0] * temp;
+
+                                let eval_0 = if d < D - 1 {
+                                    eq_evals_at_idx[0].0 * factor * after_idx_evals[D - d - 2].0
+                                } else {
+                                    eq_evals_at_idx[0].0 * factor
+                                };
+
+                                let eval_1 = if d < D - 1 {
+                                    eq_evals_at_idx[0].1 * factor * after_idx_evals[D - d - 2].1
+                                } else {
+                                    eq_evals_at_idx[0].1 * factor
+                                };
```

Comment:
It looks like `eq_evals_at_idx[0].1` is always 0, so we can skip this computation entirely